### PR TITLE
feat: update API to expose an AssetSource and remove dependency on aws-cdk-lib

### PR DESCRIPTION
**Pull request by madeline-k**
Changing the API to implement the AssetSource interface, and also not use any APIs from aws-cdk-lib. This is so that aws-cdk-lib can take a direct dependency on this package, and we can avoid a breaking change for customers using the existing awscli version.

### CODE REVIEW

File: .projenrc.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ const project = new awscdk.AwsCdkConstructLibrary({
   releaseTagPrefix: `awscli-v${MAJOR_VERSION}`,
   releaseWorkflowName: releaseWorkflowName,
   defaultReleaseBranch: defaultReleaseBranchName,
+  deps: ['@aws-cdk/interfaces'],
```
Comment (rix0rrr) [inline .projenrc.js]: Can be a devDep

**Review by rix0rrr — COMMENTED** [review]
Shouldn't this contain changes to `package.json` ?

File: src/awscli-asset.ts [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+import { ILambdaLayerAsset } from '@aws-cdk/interfaces';
 import { FileSystem } from 'aws-cdk-lib';
 import * as s3_assets from 'aws-cdk-lib/aws-s3-assets';
 import { Construct } from 'constructs';
 
 /**
  * A CDK Asset construct that contains the AWS CLI.
  */
-export class AwsCliAsset extends s3_assets.Asset {
+export class AwsCliAsset extends s3_assets.Asset implements ILambdaLayerAsset {
```
Comment (rix0rrr) [inline src/awscli-asset.ts]: Maybe call this `AwsCliV1Asset` or something.

Comment (madeline-k) [reply src/awscli-asset.ts]: The package it's released in is called `@aws-cdk/asset-awscli-v1`. So I don't think that's necessary. Let me know if you still disagree.

File: .projenrc.js (hunk 1 above)
Comment (madeline-k) [reply .projenrc.js]: But we need to expose an `ILambdaLayerAsset` in this packages API, so that makes it a direct dep, right?

Comment (rix0rrr) [reply .projenrc.js]: Hmm for the purposes of jsii, I think you are right, yes. Never mind then.

File: src/awscli-asset.ts (hunk 2 above)
Comment (rix0rrr) [reply src/awscli-asset.ts]: I know, but I like the explicitness in the final code.

Comment (madeline-k) [reply src/awscli-asset.ts]: For the `kubectl` scenario, wouldn't it be nice to be able to just change your dependency & import, and not your source code in order to upgrade?

File: .projenrc.js (hunk 1 above)
Comment (madeline-k) [reply .projenrc.js]: I temporarily removed this until this PR https://github.com/aws/aws-cdk/pull/22538 Is released anyway.

File: .projenrc.js [hunk 3]
```diff
@@ -53,6 +53,13 @@ const project = new awscdk.AwsCdkConstructLibrary({
   },
 });
 
+// We only need aws-cdk-lib and constructs for testing. Neither library is used
+// in the public API.
+project.deps.removeDependency('constructs', DependencyType.PEER);
+project.deps.addDependency('constructs@10.0.5', DependencyType.DEVENV);
```
Comment (madeline-k) [inline .projenrc.js:59]: I used explicit versions here to prevent projen from auto-upgrading to the latest. We can make that update in a separate PR.
